Code/Framework/AzCore/Platform/Common/WinAPI/AzCore/Utils/Utils_WinAPI.cpp

```diff
@@ -29,7 +30,8 @@ namespace AZ
             result.m_pathIncludesFilename = true;
             // Platform specific get exe path: http://stackoverflow.com/a/1024937
             // https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulefilenamea
-            const DWORD pathLen = GetModuleFileNameA(nullptr, exeStorageBuffer, static_cast<DWORD>(exeStorageSize));
+            wchar_t pathBufferW[AZ::IO::MaxPathLength] = { 0 };
+            const DWORD pathLen = GetModuleFileNameW(nullptr, pathBufferW, static_cast<DWORD>(exeStorageSize));
             const DWORD errorCode = GetLastError();
             if (pathLen == exeStorageSize && errorCode == ERROR_INSUFFICIENT_BUFFER)
```

Comment:
So there is a bit of an issue here.
The length returned from the `GetModuleFileNameA` does not necessarily correspond to the number of characters that will be stored in the `exeStorageBuffer`.
We can't know that information until we convert the wide string to a regular string.

That is why it may be good here to add the function that determines how many bytes are needed to for storing a converted utf16 string to a utf8 string to unchecked.h
```
static constexpr size_t utf8_codepoint_length(AZ::u32 cp)
{
    if (cp < 0x80)
    {
        return 1
    }
    else if (cp < 0x800)
    {
        return 2;
    }
    else if (cp < 0x10000)
    {
        return 3;
    }

    return 4;
}

template <typename u16bit_iterator>
size_t utf16ToUtf8BytesRequired(u16bit_iterator start, u16bit_iterator end)
{
    size_t bytesRequired = 0;
    while (start != end)
    {
        AZ::u32 cp = Utf8::Internal::mask16(*start++);
        // Take care of surrogate pairs first
        if (Utf8::Internal::is_lead_surrogate(cp))
        {
            AZ::u32 trail_surrogate = Utf8::Internal::mask16(*start++);
            cp = (cp << 10) + trail_surrogate + Internal::SURROGATE_OFFSET;
        }
        bytesRequired += utf8_codepoint_length(cp);
    }
    return bytesRequired;
}

template <typename u32bit_iterator>
size_t utf32ToUtf8BytesRequired(u32bit_iterator start, u32bit_iterator end)
{
    size_t bytesRequired = 0;
    while (start != end)
    {
        bytesRequired += utf8_codepoint_length(*start++);
    }
    return bytesRequired;
}

```

Reply:
Pretty much what can be done is to invoke `utf16ToUtf8BytesRequired` after calling the `GetModuleFileNameW` function to determine if the path will fit within the buffer.

````
const DWORD pathLen = GetModuleFileNameW(nullptr, pathBufferW, static_cast<DWORD>(AZStd::size(pathBufferW)));
const DWORD errorCode = GetLastError();
if (pathLen == exeStorageSize && errorCode == ERROR_INSUFFICIENT_BUFFER)
{
    result.m_pathStored = ExecutablePathResult::BufferSizeNotLargeEnough;
}
else if (pathLen == 0 && errorCode != ERROR_SUCCESS)
{
    result.m_pathStored = ExecutablePathResult::GeneralError;
}
else
{
    size_t utf8PathSize = utf16ToUtf8BytesRequired(pathBufferW, pathBufferW + pathLen);
    if (utf8PathSize >= exeStorageSize)
    {
        result.m_pathStored = ExecutablePathResult::BufferSizeNotLargeEnough;
    }
    else
    {
        AZStd::to_string(exeStorageBuffer, exeStorageSize, pathBufferW);
    }
}
````

Reply:
mmm, was trying to get away with doing that because it has to walk the string an extra time... but I dont see how else to solve this...  will add